fluent/src/langneg.js

```diff
@@ -0,0 +1,246 @@
+
+/**
+ * Likely subtags map from a locale in minimized form to its maximized form.
+ * It's useful for non-intuitive matches like 'en' -> 'en-US' or marking
+ * what's the languge default region and script.
+ */
+const likelySubtags = {
+  'en': 'en-Latn-US',
+  'sr': 'sr-Cyrl-RS',
+  'sr-RU': 'sr-Latn-RU',
+  'az': 'az-Latn-AZ',
+  'az-IR': 'az-Arab-IR',
+  'zh': 'zh-Hans-CN',
+  'zh-GB': 'zh-Hant-GB',
+  'zh-US': 'zh-Hant-US',
+};
+
+/**
+ * Attempts to build a most likely maximum version of the locale id based
+ * on it's minimal version.
+ *
+ * If possible it uses the lookup table above. If this one is missing,
+ * it only attempts to place the language code in the region position.
+ */
+function addLikelySubtags(locale) {
+  if (likelySubtags.hasOwnProperty(locale)) {
+    return likelySubtags[locale];
+  }
+  const loc = parseLocale(locale, true);
+  if (loc[2] === '*') {
+    loc[2] = loc[0].toUpperCase();
+    return loc.join('-');
+  }
```

Comment:
I find the line about "a language tag is a script tag with bad casing" pretty revolting, tbh. Let's not set precedent for that, people look at mozilla for how to handle standards, and this is just not like it.

I also start to realize that I actually disagree with the way you use of likelySubtags. In that context, I think most of the unfiltered likelySubtags data goes away. I'll take that top-level.